abci/client/socket_client.go

```diff
@@ -417,6 +427,23 @@ func (cli *socketClient) ApplySnapshotChunkSync(
 	return reqres.Response.GetApplySnapshotChunk(), cli.Error()
 }
 
+func (cli *socketClient) PrepareProposalSync(
+	req types.RequestPrepareProposal,
+) (*types.ResponsePrepareProposal, error) {
+
+	reqres := cli.queueRequest(types.ToRequestPrepareProposal(req))
+	return reqres.Response.GetPrepareProposal(), nil
```

Comment:
In the socket client, ABCI Requests are sent through the socket via a buffered writer.

This `PrepareProposalSync` implementation does not flush the buffered writer, so there is the possibility that the request fits in the buffer, and the buffer is not flushed.

In this case the request would wait in the outgoing buffer until the socket client's periodic flushing mechanism kicks in (line 143 of this file). Since `PrepareProposalSync` is synchronous, the caller will be blocked all that time.

To fix this, please check how existing ABCI Sync calls are implemented above (e.g. check lines 423-427)

Reply:
Resolved by https://github.com/celestiaorg/celestia-core/pull/902/commits/3c0dd40d84ba5608b11f3f3c617d4c5c90b12473